Framework/CurveFitting/src/Algorithms/DoublePulseFit.cpp

```diff
@@ -0,0 +1,406 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidCurveFitting/Algorithms/DoublePulseFit.h"
+#include "MantidCurveFitting/CostFunctions/CostFuncFitting.h"
+
+#include "MantidAPI/CompositeFunction.h"
+#include "MantidAPI/FuncMinimizerFactory.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFuncMinimizer.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/TableRow.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidCurveFitting/Functions/DeltaFunction.h"
+
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/StartsWithValidator.h"
+
+#include <math.h>
+#include <memory>
+
+using namespace Mantid::CurveFitting::Functions;
+
+namespace Mantid {
+namespace CurveFitting {
+namespace Algorithms {
+
+void setMultiDataProperties(const Mantid::API::IAlgorithm &qensFit,
```

Comment:
do we mean `qensFit`?

Reply:
is there similar code for QENS?

Reply:
Holdover from when I took the code from the QENS code. Have renamed to fittingAlgorithm. There is similar code in QENS as you need to solve the same problem there of setting up a fitting algorithm with multiple input workspaces.

Reply:
should we refactor both to share the same code then?

Reply:
Potentially yes, I've created an issue to do so here #28882